CMakeLists.txt

```diff
@@ -0,0 +1,428 @@
+option(FREERTOS_PORT_IAR_RX_V2 "FreeRTOS port (IAR) for Renesas RX v2")
+option(FREERTOS_PORT_IAR_STR71X "FreeRTOS port (IAR) for STR71x")
+option(FREERTOS_PORT_IAR_STR75X "FreeRTOS port (IAR) for STR75x")
+option(FREERTOS_PORT_IAR_STR91X "FreeRTOS port (IAR) for STR91x")
+option(FREERTOS_PORT_IAR_V850ES_FX3 "FreeRTOS port (IAR) for Renesas V850ES/Fx3")
+option(FREERTOS_PORT_IAR_V850ES_HX3 "FreeRTOS port (IAR) for Renesas V850ES/Hx3")
+option(FREERTOS_PORT_MIKROC_ARM_CM4F "FreeRTOS port (MikroC) for ARM Cortex-M4 with FPU")
+option(FREERTOS_PORT_MPLAB_PIC18F "FreeRTOS port (MPLAB) for PIC18F")
+option(FREERTOS_PORT_MPLAB_PIC24 "FreeRTOS port (MPLAB) for PIC24")
+option(FREERTOS_PORT_MPLAB_PIC32MEC14XX "FreeRTOS port (MPLAB) for PIC32MEC14xx")
+option(FREERTOS_PORT_MPLAB_PIC32MX "FreeRTOS port (MPLAB) for PIC32MX")
+option(FREERTOS_PORT_MPLAB_PIC32MZ "FreeRTOS port (MPLAB) for PIC32MZ")
+option(FREERTOS_PORT_MSVC_MINGW "FreeRTOS port for x86 MSVC or MinGW")
+option(FREERTOS_PORT_OWATCOM_16BIT_DOS_FLSH186 "FreeRTOS port (Open Watcom) for 16 bit DOS Flsh186")
+option(FREERTOS_PORT_OWATCOM_16BIT_DOS_PC "FreeRTOS port (Open Watcom) for 16 bit DOS PC")
+option(FREERTOS_PORT_PARADIGM_TERN_EE_LARGE "FreeRTOS port (Paradigm) for Tern EE large")
+option(FREERTOS_PORT_PARADIGM_TERN_EE_SMALL "FreeRTOS port (Paradigm) for Tern EE small")
+option(FREERTOS_PORT_RENESAS_RX100 "FreeRTOS port (Renesas) for RX100")
+option(FREERTOS_PORT_RENESAS_RX200 "FreeRTOS port (Renesas) for RX200")
+option(FREERTOS_PORT_RENESAS_RX600 "FreeRTOS port (Renesas) for RX600")
+option(FREERTOS_PORT_RENESAS_RX600_V2 "FreeRTOS port (Renesas) for RX600 v2")
+option(FREERTOS_PORT_RENESAS_RX700_V3_DPFPU "FreeRTOS port (Renesas) for RX700 v3 with DPFPU")
+option(FREERTOS_PORT_RENESAS_SH2A_FPU "FreeRTOS port (Renesas) for SH2A with FPU")
+option(FREERTOS_PORT_ROWLEY_MSP430F449 "FreeRTOS port (Rowley) for MSP430F449")
+option(FREERTOS_PORT_RVDS_ARM_CA9 "FreeRTOS port (RVDS) for ARM Cortex-A9")
+option(FREERTOS_PORT_RVDS_ARM_CM0 "FreeRTOS port (RVDS) for ARM Cortex-M0")
+option(FREERTOS_PORT_RVDS_ARM_CM3 "FreeRTOS port (RVDS) for ARM Cortex-M3")
+option(FREERTOS_PORT_RVDS_ARM_CM4_MPU "FreeRTOS port (RVDS) for ARM Cortex-M4 with MPU")
+option(FREERTOS_PORT_RVDS_ARM_CM4F "FreeRTOS port (RVDS) for ARM Cortex-M4 with FPU")
+option(FREERTOS_PORT_RVDS_ARM_CM7 "FreeRTOS port (RVDS) for ARM Cortex-M7")
+option(FREERTOS_PORT_RVDS_ARM7_LPC21XX "FreeRTOS port (RVDS) for ARM7 LPC21xx")
+option(FREERTOS_PORT_SDCC_CYGNAL "FreeRTOS port (SDCC) for Cygnal")
+option(FREERTOS_PORT_SOFTUNE_MB91460 "FreeRTOS port (Softune) for MB91460")
+option(FREERTOS_PORT_SOFTUNE_MB96340 "FreeRTOS port (Softune) for MB96340")
+option(FREERTOS_PORT_TASKING_ARM_CM4F "FreeRTOS port (Tasking) for ARM Cortex-M4 with FPU")
+option(FREERTOS_PORT_CDK_THEAD_CK802 "FreeRTOS port (CDK) for T-head CK802")
+option(FREERTOS_PORT_XCC_XTENSA "FreeRTOS port (XCC) for Xtensa")
+option(FREERTOS_PORT_WIZC_PIC18 "FreeRTOS port (WizC) for PIC18")
```

Comment:
Rather than creating an option per port, consider using a simple variable which the user sets. I'm in favor of the variable being set to the <compiler>_<port>. I see a couple advantages to this.

1. It becomes obvious which build you're executing
2. Prevents users from building for multiple ports at the same time

Something like the following variable assignment + conditional-expressions later on should work...
```
set( FREERTOS_PORT_SELECTED GCC_ARM_CM4F)
...
$<$<STREQUAL:${FREERTOS_PORT_SELECTED},GCC_ARM_CM4F>:portable/GCC/ARM_CM4F/port.c>

```

I realize this would allow users to put in bad compiler+port options, but a guard could be added later if that's really a problem.

Reply:
I was considering this. The big disadvantage of this approach is that it becomes unclear to the user what strings can be set to the `FREERTOS_PORT`.
It could lead to misunderstanding when `FREERTOS_PORT` is set, but its value match with none of available ports.

I will think what we can do with this..

Reply:
Fixed.